### PR TITLE
Origin/0.1a10

### PR DESCRIPTION
**Pull request by ysig**
General bugfix and enhancement addressing multiple issues: #71 #75 #90 #94 #95 #96 #97.

### CODE REVIEW

File: grakel/utils.py [hunk 1]
```diff
@@ -679,3 +688,108 @@ def get_el(ef, g):
...
+        group_id = (data.batch).unique_consecutive().tolist()
+        lookup = {i: data.batch[i].item() for i in range(data.batch.shape[0])}
+        for i, e in enumerate(data.edge_index.t().tolist()):
+            e = tuple(e)
+            group = lookup[e[0]]
+            if lookup[e[0]] != lookup[e[1]]:
+                raise ValueError('Both nodes from an edge should correspond to the same graph.')
+            edges[group].append(e)
+            if  data.edge_attr is not None:
+                edge_labels[group][e] = one_hot_edge(data.edge_attr[i])
```
Comment (giannisnik) [inline grakel/utils.py:773]: This should be replaced with:
```
edges = defaultdict(list)
node_labels = None
edge_labels = None if data.edge_attr is None else defaultdict(dict)
group_id, counts = (data.batch).unique_consecutive(return_counts=True)
group_id = group_id.tolist()
cumsum_n_nodes = counts.cumsum(0).tolist()
cumsum_n_nodes.insert(0, 0)
lookup = {i: data.batch[i].item() for i in range(data.batch.shape[0])}
for i, e in enumerate(data.edge_index.t().tolist()):
    group = lookup[e[0]]
    if lookup[e[0]] != lookup[e[1]]:
        raise ValueError('Both nodes from an edge should correspond to the same graph.')
    e = (e[0]-cumsum_n_nodes[group], e[1]-cumsum_n_nodes[group])
    edges[group].append(e)
    if  data.edge_attr is not None:
        edge_labels[group][e] = one_hot_edge(data.edge_attr[i])
```

File: grakel/utils.py [hunk 2]
```diff
@@ -679,3 +688,108 @@ def get_el(ef, g):
...
+
+        if data.x is not None:
+            node_labels = defaultdict(dict)
+            for i in range(data.x.shape[0]):
+                node_labels[data.batch[i].item()][i] = one_hot_node(data.x[i])
```
Comment (giannisnik) [inline grakel/utils.py:778]: This should be replaced:
`node_labels[data.batch[i].item()][i-cumsum_n_nodes[lookup[i]]] = one_hot_node(data.x[i])`

**Review by giannisnik — CHANGES_REQUESTED** [review]

File: grakel/utils.py (hunk 1 above)
Comment (ysig) [reply grakel/utils.py:779]: it was a bug of the __repr__
- do we really need to remap them?
Imagine a user wants to retrieve them by name.

Comment (ysig) [reply grakel/utils.py:779]: @giannisnik

Comment (giannisnik) [reply grakel/utils.py:779]: If the key of the dictionary is just the name of the node, and there is already a mapping from node names to node indices, you are right, we do not need to map them.

**Comment by ysig** [comment]
@giannisnik can you provide some tests for the new util `torch_geometric` util function at
`grakel\tests\test_utils.py`

File: grakel/utils.py (hunk 1 above)
Comment (ysig) [reply grakel/utils.py:779]: yes it was just a printing problem of the __repr__ you can try again :))

**Pull request merged**